src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+                'holder': self.account_holder,
+                'transaction_type': Keyword.BUY.value if float(record[_AMOUNT]) > 0 else Keyword.SELL.value,
+                'crypto_fee': '0' if is_fiat_asset else record[_FEE],
+                'fiat_fee': record[_FEE] if is_fiat_asset else None,
+            })
+
+            if record[_TYPE] == _TRADE and 'USD' not in record[_ASSET]:
+                kwargs.update({
+                    'spot_price': trade_history[record[_REFID]][_PRICE],
+                })
+                if float(record[_AMOUNT]) > 0:
+                    kwargs.update({
+                        'crypto_in': str(amount),
+                        'fiat_in_no_fee': str(float(trade_history[record[_REFID]][_COST]) - float(
+                            trade_history[record[_REFID]][_FEE])),
+                        'fiat_in_with_fee': trade_history[record[_REFID]][_COST],
+                    })
+                    result.append(InTransaction(**kwargs))
+                else:
+                    kwargs.update({
+                        'crypto_out_no_fee': str(amount),
+                        'crypto_out_with_fee': str(amount + float(record[_FEE])),
+                        'fiat_out_no_fee': str(float(trade_history[record[_REFID]][_COST]) - float(
+                            trade_history[record[_REFID]][_FEE])),
+                        'is_spot_price_from_web': False,
+                    })
+                    result.append(OutTransaction(**kwargs))
+            elif record[_TYPE] == _MARGIN or record[_TYPE] == _ROLLOVER:
+                kwargs.update({
+                    'transaction_type': Keyword.SELL.value,
+                    'crypto_out_no_fee': str(amount),
+                    'crypto_out_with_fee': str(amount + float(record[_FEE])),
+                    'fiat_out_no_fee': str(float(trade_history[record[_REFID]][_COST]) - float(
+                        trade_history[record[_REFID]][_FEE])),
+                    'is_spot_price_from_web': False,
+                })
+                result.append(OutTransaction(**kwargs))
+            elif record[_TYPE] == _TRANSFER:
+                kwargs.update({
+                    'transaction_type': Keyword.BUY.value,
+                    'crypto_in': str(amount),
+                    'fiat_in_no_fee': '0',
+                    'fiat_in_with_fee': '0',
+                })
+                result.append(InTransaction(**kwargs))
+            elif record[_TYPE] == _SETTLED:
+                # ignorable in terms of in/out/intra
+                pass
+            elif record[_TYPE] == _CREDIT:
+                # 'credit not implemented'
+                pass
+            elif record[_TYPE] == _STAKING:
+                # 'credit not implemented'
+                pass
+            elif record[_TYPE] == _SALE:
+                # 'credit not implemented'`
+                pass
+            else:
+                raise BaseException(f"Unimplemented=record_type{record[_TYPE]}")
```

Comment:
For now just use Exception. We're in the process of adding a new RP2RuntimeError exception that should be used in cases like this.

Reply:
RP2RuntimeError has been merged (see #117)